packages/vats/src/proposals/localchain-proposal.js

```diff
@@ -0,0 +1,112 @@
+// @ts-check
+import { E } from '@endo/far';
+import { BridgeId as BRIDGE_ID } from '@agoric/internal';
+
+/**
+ * @param {BootstrapPowers & {
+ *   consume: {
+ *     loadCriticalVat: VatLoader<any>;
+ *     bridgeManager: import('../types').BridgeManager;
+ *     localchainBridgeManager: import('../types').ScopedBridgeManager;
+ *   };
+ *   produce: {
+ *     localchain: Producer<any>;
+ *     localchainVat: Producer<any>;
+ *     localchainBridgeManager: Producer<any>;
+ *   };
+ * }} powers
+ * @param {object} options
+ * @param {{ localchainRef: VatSourceRef }} options.options
+ *
+ * @typedef {{
+ *   localchain: ERef<import('../vat-localchain.js').LocalChainVat>;
+ * }} LocalChainVats
+ */
+export const setupLocalChainVat = async (
+  {
+    consume: {
+      loadCriticalVat,
+      bridgeManager: bridgeManagerP,
+      localchainBridgeManager: localchainBridgeManagerP,
+    },
+    produce: { localchainVat, localchain, localchainBridgeManager },
+  },
+  options,
+) => {
+  const bridgeManager = await bridgeManagerP;
+  if (!bridgeManager) {
+    console.error('No bridgeManager, skipping setupLocalChainVat');
+    return;
+  }
```

Comment:
I'm cautious about failing silently. Did some CI fail without this?

Reply:
makes sense

Reply:
Erk.  Unfortunately, the sim-chain configuration of `ag-solo` uses the same config as the real chain, but lacks a bridgeManager.  I'm reverting the throw.

```
Run cd packages/solo && yarn test | $TEST_COLLECT
yarn run v1.22.19
$ ava
TAP version 13
# connecting.........................Loading slog sender modules: @agoric/telemetry/src/flight-recorder.js
................Loading slog sender modules: @agoric/telemetry/src/flight-recorder.js
.oo2024-02-21T21:[15](https://github.com/Agoric/agoric-sdk/actions/runs/7995614069/job/21836438318#step:5:16):49.920Z SwingSet: xsnap: v1: Error#1 ERROR_NOTE: Sent as error:liveSlots:v1#70001
2024-02-21T21:15:49.921Z SwingSet: xsnap: v1: Error#1 ERROR_NOTE: Sent as error:liveSlots:v1#70002
2024-02-21T21:15:49.922Z SwingSet: kernel: ##### KERNEL PANIC: kp40.policy panic: rejected {"body":"#{\"#error\":\"No bridgeManager, cannot setupLocalChainVat\",\"errorId\":\"error:liveSlots:v1#70002\",\"name\":\"Error\"}","slots":[]} #####
2024-02-21T21:15:49.936Z SwingSet: kernel: ##### KERNEL PANIC: kp40.policy panic: rejected {"body":"#{\"#error\":\"vat terminated\",\"name\":\"Error\"}","slots":[]} #####
2024-02-21T21:15:49.936Z SwingSet: kernel: ##### KERNEL PANIC: critical vat v1 failed #####
[20](https://github.com/Agoric/agoric-sdk/actions/runs/7995614069/job/21836438318#step:5:21)24-02-[21](https://github.com/Agoric/agoric-sdk/actions/runs/7995614069/job/21836438318#step:5:22)T21:15:49.937Z block-manager: AG_COSMOS_INIT error: (Error#1)
Error#1: #{"#error":"No bridgeManager, cannot setupLocalChainVat","errorId":"error:liveSlots:v1#70001","name":"Error"}
```

Reply:
Gotcha. A comment as to why would help future readers